### PR TITLE
cast updated fields

### PR DESCRIPTION
**Pull request by trinity-1686a**
### Description

Following a docmapper update, we might end up with documents with a one type stored, while the docmapper expecting a different type, attempt to cast what we have to what we want.

### How was this PR tested?

completed unit tests, tested manually after updating an index

### CODE REVIEW

**Comment by fulmicoton** [comment]
Make sure the motivation also ends up in the commit description.

**Review by guilload — APPROVED** [review]

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/mapping_tree.rs [hunk 1]
```diff
@@ -385,48 +385,166 @@ fn extract_json_val(
...
+            // TODO we could cast str to bytes
             let json_value = bytes_options.output_format.format_to_json(bytes);
             Some(json_value)
         }
-        (TantivyValue::Date(date_time), LeafType::DateTime(date_time_options)) => {
+        (_, LeafType::DateTime(date_time_options)) => {
+            let date_time = match date_time_options.reparse_tantivy_value(&value) {
+                Some(date_time) => date_time,
+                None => {
+                    warn!("TODO");
```
Comment (guilload) [inline quickwit/quickwit-doc-mapper/src/default_doc_mapper/mapping_tree.rs:527]: rate_limited warn?

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/mapping_tree.rs [hunk 2]
```diff
@@ -385,44 +384,219 @@ fn extract_json_val(
...
+        TantivyValue::Bool(b) => Some(*b as i64),
+        _ => None,
+    }
+    .and_then(|u64_val| u64_val.to_json(numeric_options.output_format))
+    .ok_or(value)
+}
+
+/// Transform a tantivy object into a serde_json one, without cloning strings.
+/// It still allocates maps.
+// TODO we should probably move this to tantivy, it has the opposite conversion already
```
Comment (guilload) [inline quickwit/quickwit-doc-mapper/src/default_doc_mapper/mapping_tree.rs:507]: @PSeitz, can we move that code to tantivy eventually?

**Review by guilload — APPROVED** [review]

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/mapping_tree.rs [hunk 3]
```diff
@@ -385,44 +384,219 @@ fn extract_json_val(
...
 }
 
+fn value_to_string(value: TantivyValue) -> Result<JsonValue, TantivyValue> {
+    match value {
+        TantivyValue::Str(s) => return Ok(JsonValue::String(s)),
+        TantivyValue::U64(number) => Some(number.to_string()),
+        TantivyValue::I64(number) => Some(number.to_string()),
+        TantivyValue::F64(number) => Some(number.to_string()),
+        TantivyValue::Bool(b) => Some(b.to_string()),
+        TantivyValue::Date(ref date) => {
```
[the remark this answers is not in the document]
Comment (trinity-1686a) [reply quickwit/quickwit-doc-mapper/src/default_doc_mapper/mapping_tree.rs:394]: hum, yeah you're right

**Pull request merged**